src/Symfony/Component/Routing/RouteCompiler.php

```diff
@@ -223,14 +223,37 @@ private static function compilePattern(Route $route, $pattern, $isHost)
         }
 
         return array(
-            'staticPrefix' => 'text' === $tokens[0][0] ? $tokens[0][1] : '',
+            'staticPrefix' => self::determineStaticPrefix($route, $tokens),
```

Comment:
This change allows for the longest possible static prefix. The separator leading up to the expression is only optional when the parameter has a default. So that means we can add it to the static prefix when is doesn't have a default, leading to better prefix groups.